staging/src/k8s.io/apiserver/pkg/server/routine/routine.go

```diff
@@ -35,6 +35,20 @@ func WithTask(parent context.Context, t *Task) context.Context {
 	return request.WithValue(parent, taskKey, t)
 }
 
+// AppendTask appends a task executed after completion of existing task.
+// It is a no-op if there is no existing task.
+func AppendTask(ctx context.Context, t *Task) bool {
```

Comment:
This deserves some unit test.

Reply:
Added

Reply:
You added unit test, but it's not testing the logic in this function.

I would like to see a unit test for the Append function itself.

Reply:
I added `TestAppendTest`. Can you please clarify which logic is missed in that test?

https://github.com/kubernetes/kubernetes/pull/125626/files#diff-6fb1ba0a2fe370754b5df8125bd60ba5523c1266d4526f803a21eaee26db8d0eR47

Reply:
It doesn't call "AppendTask" function. I want the unit test to test exactly this function.

Reply:
Make sense. Changed.